browser/app/js/browser/MainActions.js

```diff
@@ -33,8 +33,20 @@ export const MainActions = ({
   )
   const onFileUpload = e => {
     e.preventDefault()
-    uploadFile(e.target.files[0])
-    e.target.value = null
+    let files = e.target.files
+    let filesToUploadCount = files.length
+    let currentFileIndex = 0
+    const uploadNextFile = function() {
+      if (currentFileIndex < filesToUploadCount) {
+        let file = files.item(currentFileIndex++)
+        if (file) {
+          let nextFilePossible = currentFileIndex < filesToUploadCount
+          let callback = nextFilePossible ? uploadNextFile : null
+          uploadFile(file, callback)
```

Comment:
@DQvsRA is there any problem in uploading the files in parallel?

Reply:
@kmkanagaraj Thank for the reply. I don't know, maybe, I need to make a deeper investigation but first commit was done with parallel uploading. But why not to have it in sequenced order?

Reply:
@DQvsRA AbortModal and Upload progress can handle parallel uploads. So we can actually upload in parallel instead of one after another. If you are facing any issue, please let me know, i can take a look.

Reply:
```
for (let i = 0; i < filesToUploadCount; i++) {
  uploadFile(files.item(i))
}
```
The above code should work fine.

Reply:
@kmkanagaraj Thanks, I will revert to your approach and revert all the changes